packages/admin-ui/src/modal/modal-button.tsx

```diff
@@ -1,26 +1,20 @@
 import type { Ref } from 'react'
 import React, { forwardRef } from 'react'
-import { cx, csx } from '@vtex/admin-ui-core'
-import { DialogDismiss, Button } from 'ariakit'
+import { useDialogDismiss } from 'ariakit'
 
 import type { ButtonProps } from '../button'
-import { useButton } from '../button'
-import type { AnyObject } from '@vtex/admin-ui-util'
+import { Button } from '../button'
 
 const ModalButton = forwardRef(
   (props: ModalButtonProps, ref: Ref<HTMLButtonElement>) => {
-    const { className = '', dismissModal = false, ...htmlProps } = props
-    const { baseStyle = {}, ...buttonHtmlProps } = useButton(htmlProps)
+    const { dismissModal = false, ...restProps } = props
 
-    const buttonProps: AnyObject = {
-      ref,
-      className: cx(csx(baseStyle), className),
-      ...buttonHtmlProps,
-    }
+    const { children, ...dismissProps } = useDialogDismiss(restProps)
+    const buttonProps = dismissModal
+      ? { ...dismissProps, children: restProps.children }
+      : restProps
```

Comment:
Doesn't work. I tested it

Reply:
The thing is, it does overwrite. The thing is when there's no children prop from the user, the dismissProps ends up with a children prop that is just the icon "X". So for example in a button without children but with icon (case of the close button in the modal) the button was displaying 2 "x"s. The one from icon, and the one from children generated by dismissprops

This problem was not here before because the buttonprops was generated before so it overwrote "children" with it's own children